internal/controller/eventing/controller.go

```diff
@@ -116,12 +132,112 @@ func (r *Reconciler) handleEventingDeletion(_ context.Context, _ *eventingv1alph
 }
 
 func (r *Reconciler) handleEventingReconcile(ctx context.Context,
-	_ *eventingv1alpha1.Eventing, log *zap.SugaredLogger) (ctrl.Result, error) {
+	eventing *eventingv1alpha1.Eventing, log *zap.SugaredLogger) (ctrl.Result, error) {
 	log.Info("handling Eventing reconciliation...")
-	// TODO: Implement me.
 
-	// just to use the variables.
-	log.Info(FinalizerName, ManagedByLabelKey, ManagedByLabelValue)
+	// set state processing if not set yet
+	r.InitStateProcessing(eventing)
+
+	for _, backend := range eventing.Spec.Backends {
+		switch backend.Type {
+		case eventingv1alpha1.NatsBackendType:
+			return r.reconcileNATSBackend(ctx, eventing, log)
+		case eventingv1alpha1.EventMeshBackendType:
+			return r.reconcileEventMeshBackend(ctx, eventing, log)
+		default:
+			return ctrl.Result{Requeue: false}, fmt.Errorf("not supported backend type %s", backend.Type)
+		}
+	}
+	// this should never happen, but if happens do nothing
+	return ctrl.Result{}, nil
```

Comment:
Even if this might not happen, how about we return:

```golang
return ctrl.Result{Requeue: false}, fmt.Errorf("backend is not provided in the spec")
```